os/pm/pm_update.c

```diff
@@ -110,6 +111,38 @@ static const int16_t g_pmcoeffs[CONFIG_PM_MEMORY - 1] = {
 };
 #endif
 
+#ifdef CONFIG_IDLE_PM
+/* Threshold time slice count to enter the next low power consdumption
+ * state. Indexing is next state 0:IDLE, 1: STANDBY, 2: SLEEP.
+ */
+
+static const uint16_t pm_sleepstate_count[CONFIG_ARCH_HAVE_IDLE_PM_NSTATE] = {
+	CONFIG_PM_STATE1_ENTER_COUNT,
+	(CONFIG_PM_STATE1_ENTER_COUNT * 2),
+	(CONFIG_PM_STATE1_ENTER_COUNT * 3)
+};
+
+/* Threshold activity values to enter into the next lower power consumption
+ * state. Indexing is next state 0:IDLE, 1:STANDBY, 2:SLEEP.
+ */
+
+static const uint16_t pm_sleepstate_enter_threshold[CONFIG_ARCH_HAVE_IDLE_PM_NSTATE] = {
+	CONFIG_PM_STATE1_ENTER_THRESH,
+	CONFIG_PM_STATE1_ENTER_THRESH,
+	CONFIG_PM_STATE1_ENTER_THRESH,
+};
+
+/* Threshold activity values to enter into the next lower power consumption
+ * state. Indexing is next state 0:IDLE, 1:STANDBY, 2:SLEEP.
+ */
+
+static const uint16_t pm_sleepstate_exit_threshold[CONFIG_ARCH_HAVE_IDLE_PM_NSTATE] = {
+	CONFIG_PM_STATE1_EXIT_THRESH,
+	CONFIG_PM_STATE1_EXIT_THRESH,
+	CONFIG_PM_STATE1_EXIT_THRESH,
+};
+#else
+
```

Comment:
how are you sure that only 3 states will be there ? what are these exit thresholds?